semgrep-core/src/runner/Run_semgrep.ml

```diff
@@ -369,6 +369,13 @@ let rules_for_xlang xlang rules =
          | Xlang.L (x, _empty), Xlang.L (y, ys) -> List.mem x (y :: ys)
          | (Xlang.LRegex | Xlang.LGeneric | Xlang.L _), _ -> false)
 
+module Rule_table = Map.Make (String)
+
+let mk_rule_table rules =
```

Comment:
I usually use Common.hash_of_list for that instead of instantiating each time a module.